Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -905,6 +905,24 @@ protected theorem Complex.inner (w z : ℂ) : ⟪w, z⟫_ℝ = (conj w * z).re :
 
 end RCLikeToReal
 
+/-- An `RCLike` field is a real inner product space. This is not registered as an instance since it
+creates problems with the case `𝕜 = ℝ`, but it can be used in a proof to obtain a real inner product
+space structure on an `RCLike` field.
+
+This is different from the inner product space structure obtained from
+`InnerProductSpace.rclikeToReal 𝕜 𝕜` because of a diamond in the `ℝ`-normed space instances:
+`NormedSpace.restrictScalars ℝ 𝕜 𝕜` ≠ `NormedAlgebra.toNormedSpace' (𝕜 := ℝ) (𝕜' := 𝕜)`. -/
+noncomputable def RCLike.innerProductSpaceReal : InnerProductSpace ℝ 𝕜 where
+  __ := Inner.rclikeToReal 𝕜 𝕜
+  __ := NormedAlgebra.toNormedSpace' (𝕜 := ℝ) (𝕜' := 𝕜)
```

Comment:
Maybe neater as
```suggestion
  __ : NormedSpace ℝ 𝕜 := inferInstance
```

Reply:
Maybe, but that would make the docstring harder to understand

Reply:
Actually, I could just remove the line entirely